File: CS/ApplicationServerService/ServerApplication.cs
using System;
using System.Collections.Generic;
using System.Text;
using DevExpress.ExpressApp.MiddleTier;
using DevExpress.ExpressApp.Xpo;
using DevExpress.ExpressApp;
using DevExpress.ExpressApp.Security;
using SecuritySystemExample.Module;
using DevExpress.ExpressApp.Win.SystemModule;
using DevExpress.ExpressApp.Web.SystemModule;

namespace ApplicationServerService {
    public class ApplicationServerServiceServerApplication : ServerApplication {
        public ApplicationServerServiceServerApplication() {
            this.ApplicationName = "SecuritySystemExample";
            this.Modules.Add(new SystemWindowsFormsModule());
            this.Modules.Add(new SystemAspNetModule());
            this.Modules.Add(new SecuritySystemExampleModule());
            this.Modules.Add(new SecurityModule());
        }
        protected override void OnDatabaseVersionMismatch(DevExpress.ExpressApp.DatabaseVersionMismatchEventArgs args) {
            args.Updater.Update();
            args.Handled = true;
        }
        protected override void CreateDefaultObjectSpaceProvider(CreateCustomObjectSpaceProviderEventArgs args) {
            args.ObjectSpaceProvider = new XPObjectSpaceProvider(args.ConnectionString, args.Connection);
        }
    }
}

File: VB/ApplicationServerService/ServerApplication.vb
Imports System
Imports System.Collections.Generic
Imports System.Text
Imports DevExpress.ExpressApp.MiddleTier
Imports DevExpress.ExpressApp.Xpo
Imports DevExpress.ExpressApp
Imports DevExpress.ExpressApp.Security
Imports SecuritySystemExample.Module
Imports DevExpress.ExpressApp.Win.SystemModule
Imports DevExpress.ExpressApp.Web.SystemModule

Namespace ApplicationServerService
    Public Class ApplicationServerServiceServerApplication
        Inherits ServerApplication

        Public Sub New()
            Me.ApplicationName = "SecuritySystemExample"
            Me.Modules.Add(New SystemWindowsFormsModule())
            Me.Modules.Add(New SystemAspNetModule())
            Me.Modules.Add(New SecuritySystemExampleModule())
            Me.Modules.Add(New SecurityModule())
        End Sub
        Protected Overrides Sub OnDatabaseVersionMismatch(ByVal args As DevExpress.ExpressApp.DatabaseVersionMismatchEventArgs)
            args.Updater.Update()
            args.Handled = True
        End Sub
        Protected Overrides Sub CreateDefaultObjectSpaceProvider(ByVal args As CreateCustomObjectSpaceProviderEventArgs)
            args.ObjectSpaceProvider = New XPObjectSpaceProvider(args.ConnectionString, args.Connection)
        End Sub
    End Class
End Namespace
